test/ng/filter/filterSpec.js

```diff
@@ -425,6 +425,23 @@ describe('Filter: filter', function() {
       toThrowMinErr('filter', 'notarray', 'Expected array but received: {"toString":null,"valueOf":null}');
   });
 
+  it('should not throw an error if used with an array like object', function() {
+    var nodeList = jqLite("<p><span>Misko</span><span>Igor</span><span>Brad</span></p>")[0].childNodes;
+    function nodeFilterPredicate(node) {
+      return node.innerHTML.indexOf("I") !== -1;
+    }
+
+    function aux() {
+      expect(filter(arguments, 'i').length).toBe(2);
+    }
+
+
+    aux({name: 'Misko'}, {name: 'Igor'}, {name: 'Brad'});
```

Comment:
It was not immediately clear to me what was being tested here, i.e. the array-like `arguments` object. It would be better to write something like:

```js
function getArguments() {
  return arguments;
}
var argsObj = getArguments({name: 'Misko'}, {name: 'Igor'}, {name: 'Brad'});
expect(filter(argObj, 'i').length).toBe(2);
```